### ISSUE TITLE
[Feature] Add Validation for Mobile Number in Signup Form

### ISSUE THREAD

**Issue by dpgaharwal** [PratikMane0112/TelMedSphere#26]
### Issue Description:
Currently, the signup form does not validate the mobile number format (even 8 digit mobile number will proceed with signup) and only enforces a basic password length requirement (minimum of 6 characters). This leads to potential issues where users can sign up with invalid mobile numbers or weak passwords. To enhance security and data integrity, we need to:
1. Add proper validation for the mobile number format.
2. Enhance password validation to meet more robust security standards.

### Expected Behavior:

1. **Mobile Number Validation**:
   - The mobile number must be exactly 10 digits long (or as per the required format).
   - The frontend should validate the mobile number format before submission, and the backend should ensure that the mobile number is in the correct format before processing the signup request.

2. **Password Validation**:
   - The password must be at least 8 characters long.
   - The password must contain at least one uppercase letter.
   - The password must contain at least one lowercase letter.
   - The password must contain at least one numeric digit.
   - The password must contain at least one special character (e.g., `!@#$%^&*`).

3. **Frontend**:
   - If the mobile number is invalid (e.g., incorrect format or length), an error message should be displayed, and the form should not be submitted.
   - If the password does not meet the validation criteria, an error message should be displayed, and the form should not be submitted.

4. **Backend**:
   - Upon receiving the signup request, the server should verify that both the mobile number and password meet the required validation criteria.
   - If the mobile number is invalid or the password does not meet the required standards, the server should reject the request with appropriate error messages.

By implementing these validations, we will improve both the user experience and the security of the application.

Here below is the issue of the video..

https://github.com/user-attachments/assets/b6ac8930-ff28-424f-8cd5-c0123c8b62ec

### Checklist

- [X] I have checked the existing issues.
- [X] I follow [Contributing Guidelines](https://github.com/PratikMane0112/TelMedSphere/blob/master/.github/CONTRIBUTING_GUIDELINES.md) of this project.
- [X] I am a SWOC'25 contributor.
- [X] I am a KWOC'24 contributor.

### Code of Conduct

- [X] I agree to follow this project's "[Code of Conduct](https://github.com/PratikMane0112/TelMedSphere/blob/master/.github/CODE_OF_CONDUCT.md)".

**Comment by PratikMane0112** [comment]
Hey @dpgaharwal, WelCome to 💖TelMedSphere !!!!! . Please go ahead with your issue.

**Comment by dpgaharwal** [comment]
Please assign this issue to me under SWOC

**Comment by PratikMane0112** [comment]
Foe testing purpose you can use test accounts -
For patient -
username - [patient@gmail.com](mailto:patient@gmail.com)
password - [patient@gmail.com](mailto:patient@gmail.com)
For doctor -
username - [doctor@gmail.com](mailto:doctor@gmail.com)
password - [doctor@gmail.com](mailto:doctor@gmail.com)

**Comment by dpgaharwal** [comment]
@PratikMane0112 as assigned this issue today will count contribution of SWOC?

**Comment by PratikMane0112** [comment]
> @PratikMane0112 Sure I will. as assigned this issue today will count contribution of SWOC?

Yes, definitely !!!! I will merge your PR now for KWoC but for SWoC as per guidelines, I will add label of it on 1 Jan (or as per label rules which they will tell on 1 Jan)

**Comment by PratikMane0112** [comment]
For any further updaye or Discuddion join discoerd server Link in README file itself

**Comment by PratikMane0112** [comment]
#27  Next time please follow the teck-stack of codebase .....

**Comment by dpgaharwal** [comment]
Hi, @PratikMane0112 
As this issue was closed.
But I really want to contribute to this for learning purpose.
So I made a fork and add the changes and push to my forked project creating new branch would you be free to check.


My forked repo: https://github.com/PratikMane0112/TelMedSphere
Branch name: feat/Add-phone-validation

Here is the result follows the tech stack
![image](https://github.com/user-attachments/assets/18874b6d-53ac-4667-8ce9-37799cf17ef3)

![image](https://github.com/user-attachments/assets/f5ad5919-e954-452b-8b63-eaffe0b248ed)
